include/onnxruntime/core/optimizer/graph_transformer.h

```diff
@@ -5,10 +5,16 @@
 
 #include "core/common/common.h"
 #include "core/graph/graph_viewer.h"
-#include "core/optimizer/rewrite_rule.h"
 
 namespace onnxruntime {
 
+enum TransformerLevel {
+  Default = 1,
+  Optional_L1 = 2,
+  Optional_L2 = 4,
+  MaxTransformerLevel = 8,
+};
```

Comment:
nit: Seems a little strange for 'max' to not equal the maximum valid value.

Reply:
Also it seems "dangerous" if someone introduces a new level after. I think we should not be requiring this here to be dependent on the implementation of performing powers of two.

Reply:
I agree. Unless there's some scenario where the user would be able to pick say (Default | Level2), it would be simpler for this just to be a normal enum.

We can easily turn that into a set of bits via (1 << value) when creating the 'levels enabled' value.

Reply:
I agree. This enum could just have simple values. Turning those into a bitmask to use to determine if a level is enabled is easily done in the couple of internal places it's needed.

Reply:
It is very common to use enums for bit masking... I dont see why it is dangerous plus if we add proper unit tests to make sure any bad changes are caught there is no reason to worry.

However since you both think it is bad idea I am changing this to be a regular enum.